samza-test/src/test/java/org/apache/samza/test/processor/TestZkLocalApplicationRunner.java

```diff
@@ -1230,6 +1294,159 @@ public void testFirstProcessorDiesButSameRunIdForBatch() throws InterruptedExcep
     appRunner3.waitForFinish();
   }
 
+  private void writeStartpoints(StartpointManager startpointManager, String streamName, Integer partitionCount, Startpoint startpoint) {
+    for (int p = 0; p < partitionCount; p++) {
+      startpointManager.writeStartpoint(new SystemStreamPartition(TEST_SYSTEM, streamName, new Partition(p)), startpoint);
+    }
+  }
+
+  @Test
+  public void testStartpoints() throws InterruptedException {
```

Comment:
Let's move this to a separate class instead of TestZkLAR.

Reply:
@prateekm - Done. Reverted TestZkLAR (minus some minor refactoring) and moved this to its own test harness class.